crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs

```diff
@@ -404,3 +404,175 @@ struct __private__ResultVoidAndPtr __swift_bridge__$some_function(void);
         .test();
     }
 }
+
+/// Test code generation for Rust function that returns a Result<T, E> where T is a opaque Rust type and
+/// E is a transparent enum type.
+mod extern_rust_fn_return_result_opaque_rust_type_and_transparent_enum_type {
+    use super::*;
+
+    fn bridge_module_tokens() -> TokenStream {
+        quote! {
+            mod ffi {
+                extern "Rust" {
+                    type SomeOkType;
+                }
+                enum SomeErrEnum {
+                    Variant1,
+                    Variant2(i32),
+                }
+                extern "Rust" {
+                    fn some_function() -> Result<SomeOkType, SomeErrEnum>;
+                }
+            }
+        }
+    }
+
+    fn expected_rust_tokens() -> ExpectedRustTokens {
+        ExpectedRustTokens::Contains(quote! {
+            #[repr(C)]
+            pub enum ResultSomeOkTypeAndSomeErrEnum{
+                Ok(*mut super::SomeOkType),
+                Err(__swift_bridge__SomeErrEnum),
+            }
+
+
+            #[export_name = "__swift_bridge__$some_function"]
+            pub extern "C" fn __swift_bridge__some_function() -> ResultSomeOkTypeAndSomeErrEnum{
+                match super::some_function() {
+                    Ok(ok) => ResultSomeOkTypeAndSomeErrEnum::Ok(Box::into_raw(Box::new({
+                        let val: super::SomeOkType = ok;
+                        val
+                    })) as *mut super::SomeOkType),
+                    Err(err) => ResultSomeOkTypeAndSomeErrEnum::Err(err.into_ffi_repr()),
+                }
+            }
+        })
+    }
+
+    fn expected_swift_code() -> ExpectedSwiftCode {
+        ExpectedSwiftCode::ContainsAfterTrim(
+            r#"
+public func some_function() throws -> SomeOkType {
+    try {
```

Comment:
The reason that the rest of the Swift codegen puts everything on one line is so that we can avoid needing to worry about indentation.

Technically indentation doesn't matter, but it feels wrong to me for us to emit code with "incorrect" indentation.

i.e. if you have a `Result<Result<T, E>, Foo>` the indentation would start to get weird since this uses a hard coded indentation of 4 spaces.

---

**So, I'd prefer if we just made this one line.**

And if we ever wanted to generate multiple lines we could introduce an indentation tracker so that the code that we emitted was nested nicely.

---

Let me know if you disagree.